testnet/stacks-node/src/tests/signer/v0.rs

```diff
@@ -540,6 +543,9 @@ fn miner_gather_signatures() {
     signer_test.boot_to_epoch_3();
     let timeout = Duration::from_secs(30);
 
+    // give the system a chance to mine a Nakamoto block
+    sleep_ms(30_000);
```

Comment:
Is this sleep unnecessary?

Reply:
If the sleep is necessary, it should be replaced with a `wait_for()` invocation that checks for the condition the sleep is waiting on here -- I think the `blocks_processed` mutex could be used here (or invoking `get_chain_info`)

Reply:
Yes, it's necessary, and I'll replace it with `wait_for()`.  It's waiting for the Stacks chain tip to advance.